CMakeLists.txt

```diff
@@ -192,47 +192,35 @@ if (MSVC)
   add_compile_options(/wd4996)
   # Disable GDI (which we don't need, and causes some macro
   # re-definition issues if wingdi.h is included)
-  add_compile_options(/DNOGDI)
+  add_compile_definitions("NOGDI")
   # Add /MPn flag from CMake invocation (if defined).
   add_compile_options(${MSVC_MP_FLAG})
   # Build-specific flags
-  if (CMAKE_BUILD_TYPE MATCHES "Debug")
-    add_compile_options(/DDEBUG /Od /Zi /bigobj)
-  elseif (CMAKE_BUILD_TYPE MATCHES "Release")
-    add_compile_options(/DNDEBUG /Ox)
-  elseif (CMAKE_BUILD_TYPE MATCHES "RelWithDebInfo")
-    add_compile_options(/DNDEBUG /Ox /Zi)
-  endif()
+  add_compile_definitions("$<IF:$<CONFIG:Debug>,DEBUG,NDEBUG>")
+  add_compile_options(
+    "$<$<CONFIG:Debug>:/Od;/Zi;/bigobj>"
+    "$<$<CONFIG:Release,RelWithDebInfo>:/Ox>"
+    "$<$<CONFIG:RelWithDebInfo>:/Zi>")
```

Comment:
Added documentation for the MSVC options and defined `/Zi` in one place. I also added a note on why we must use generator expressions.

Regarding using `/ZI` instead of `/Zi`, [documentation says](https://learn.microsoft.com/en-us/cpp/build/reference/z7-zi-zi-debug-information-format?view=msvc-170#zi-1) that it "can cause issues in code size, performance, and compiler conformance".